src/input.c

```diff
@@ -48,20 +49,273 @@ FILE* check_args(int argc, char *argv[])
+  if(!read_clauses(fp, file_size, data, clause_lengths, unit_clauses, unit_clauses_length))
+  { return -1; }
+
+  // Load values into structs.
+  unin->data = data;
+  unin->nbclauses = nbclauses; 
+  unin->nbvars = nbvar;
+  unin->clause_lengths = clause_lengths;
+
+  in->data = unit_clauses;
+  in->length = *unit_clauses_length;
+
   LOG("INPUT PARSER RETURNING", 1);
-  return NULL;
+  return 1;
 }
 
 /*******************************************************************************************
- * NAME :             check_args
+ * NAME :             input_string
+ *
+ * DESCRIPTION :      Reads a line of data from a file.
+ *
+ * Method based on solution from 
+ * http://stackoverflow.com/questions/16870485/how-can-i-read-an-input-string-of-unknown-length
+ * Username: BLUEPIXY
+ *
+ * INPUTS :
+ *      PARAMETERS :   
+ *          FILE*       fp            the pointer to the file
+ *          size_t      size          the number of bytes to read
+ *
+ * OUTPUTS :
+ *      RETURN :
+ *          char*                      the string read from the file on success, 
+ *                                     the original string on failure
+ */
+char* input_string(FILE* fp, size_t size)
+{
+//The size is extended by the input with the value of the provisional
+    char *str;
+    int ch;
+    size_t len = 0;
+    str = realloc(NULL, size); //size is start size
+    if(!str)return str;
+    while(EOF != (ch = fgetc(fp)) && ch != '\n') // && ch != '\n'
+    { 
+        str[len++]=ch;
+        if(len==size){
+            str = realloc(str, (size+=16));
+            if(!str)return str;
+        }
+    }
+    str[len++]='\0';
+
+    return realloc(str, len);
+}
+
+/*******************************************************************************************
+ * NAME :             get_file_size
+ *
+ * DESCRIPTION :      Returns the sie of a file.
+ *
+ *
+ * INPUTS :
+ *      PARAMETERS :   
+ *          FILE*       fp            the pointer to the file
+ *
+ * OUTPUTS :
+ *      RETURN :
+ *          int                      the size of the file in bytes
+ */
+ int get_file_size(FILE* fp)
+ {
+  fseek(fp, 0L, SEEK_END);
+  int size = ftell(fp);
+  fseek(fp, 0L, SEEK_SET);
+  return size;
+ }
+
+/*******************************************************************************************
+ * NAME :             read_comments
  *
- * DESCRIPTION :      <DESCRIPTION STUBB>
+ * DESCRIPTION :      Progresses the file pointer past the commentes.
+ *
+ *
+ * INPUTS :
+ *      PARAMETERS :   
+ *          FILE*       fp            The pointer to the file.
+ *          int         file size     The size of the file in bytes.
+ *
+ * OUTPUTS :
+ *      RETURN :
+ *          char*                     The line of the next line ("problem" line).
+ */
+char* read_comments(FILE* fp, int file_size)
+{
+  char* line = input_string(fp, file_size);
+
+  while(line != NULL && strcmp(line, ""))
+  {
+    if(line[0] == 'c')
+    { /* Ignore comments */ }
```

Comment:
blocks like this can on a single line:

if (line[0] == 'c') { /* Ignore comments */ }